tests/buffer/buffer_set_final_data_1.cpp

```diff
@@ -0,0 +1,90 @@
+/* RUN: %{execute}%s
+
+   Exercise buffer::set_final_data()
+*/
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <boost/test/minimal.hpp>
+
+using namespace cl::sycl;
+
+int test_main(int argc, char *argv[]) {
+  constexpr size_t N = 16;
+  {
+    std::vector<int> v (N);
+    for(size_t i = 0; i < N; i++)
+      v[i] = i;
+    std::vector<int> w (N);
+
+    {
+      buffer<int> result {v.begin(), v.end()};
+      result.set_final_data<typename std::vector<int>::iterator >(w.begin());
+    }
+
+    for (int i = 0; i != N; ++i) {
```

Comment:
Add a comment to explain what happens, such as "Verify that w has not been changed"